astropy/table/table.py

```diff
@@ -2262,6 +2262,21 @@ def __len__(self):
             self._first_colname = next(iter(self.columns))
             return len(self.columns[self._first_colname])
 
+    def __or__(self, other):
+        if isinstance(other, Table):
+            updated_table = self.copy()
+            updated_table.update(other)
+            return updated_table
+        else:
+            return NotImplemented
```

Comment:
Adding a test for this line and the one in `ior`` should fix the coverage failures! Then this PR is probably ready to be merged. Thanks @AlinderS

Reply:
Note to the comment: How I would test this is by its effect, i.e., that `some_table | some_dict` gives a `TypeError`

Reply:
I don't understand this coverage report since the tests actually do run `t1 | 1` which I explicitly confirmed does run the `return NotImplemented` line. There is also test coverage for `t1 |= 1`. Any ideas? We can just ignore these coverage warnings but it is annoying to not understand.

Reply:
Very weird.  And serves me right for not looking properly -- continuing in that vein nevertheless (no time to test myself...), if one runs `tox -e test-coverage` on one's own computer, does it also mark it as not covered?

Anyway, I would suggest to ignore this...